src/main/java/winstone/Launcher.java

```diff
@@ -208,6 +214,22 @@ public Launcher(Map args) throws IOException {
         Runtime.getRuntime().addShutdownHook(new ShutdownHook(this));
     }
 
+    private void writePortToFileIfNeeded() throws IOException {
+        String portFileName = System.getProperty(WINSTONE_PORT_FILE_NAME_PROPERTY);
+        if (portFileName != null) {
+            Connector connector = server.getConnectors()[0];
+            if (connector instanceof ServerConnector) {
+                int port = ((ServerConnector) connector).getLocalPort();
+                File portFile = new File(portFileName);
+                File portDir = portFile.getParentFile();
+                portDir.mkdirs();
+                try (BufferedWriter writer = new BufferedWriter(new FileWriter(portFile))) {
```

Comment:
not entirely convicned you need a `BufferedWriter` here as we are writing a very small number of bytes :)